pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -57,16 +53,11 @@ func (h *Helper) GetDockerAuth(imageName, authType string) (docker.AuthConfigura
 		return docker.AuthConfiguration{}, false
 	}
 
-	var cfg credentialprovider.DockerConfig
-	var err error
-	if strings.HasSuffix(dockercfgPath, kapi.DockerConfigJsonKey) || strings.HasSuffix(dockercfgPath, "config.json") {
-		cfg, err = readDockerConfigJson(dockercfgPath)
-	} else if strings.HasSuffix(dockercfgPath, kapi.DockerConfigKey) {
-		cfg, err = readDockercfg(dockercfgPath)
-	}
-
+	dockercfgDir, dockercfgFile := filepath.Split(dockercfgPath)
+	credentialprovider.SetPreferredDockercfgPath(dockercfgDir)
```

Comment:
you should get the preferred path, store it temporarily, and reset it to the original value after the Read call, so that you are not permanently changing the preferred path value.